src/ServiceControl/Recoverability/Retrying/Handlers/PendingRetriesHandler.cs

```diff
@@ -24,23 +25,34 @@ public PendingRetriesHandler(IBus bus, IDocumentStore store, RetryDocumentManage
         }
 
         public void Handle(RetryPendingMessagesById message)
+        {
+            HandleAsync(message).GetAwaiter().GetResult();
+        }
+
+        private async Task HandleAsync(RetryPendingMessagesById message)
         {
             foreach (var messageUniqueId in message.MessageUniqueIds)
             {
-                manager.RemoveFailedMessageRetryDocument(messageUniqueId);
+                await manager.RemoveFailedMessageRetryDocument(messageUniqueId)
```

Comment:
Await the entire list instead of each message?

Reply:
See https://github.com/Particular/ServiceControl/pull/1176#discussion_r194648515